### PR TITLE
Use DataProcessingError for 1Mb limit for op batches

### PR DESCRIPTION
**Pull request by NicholasCouri**
# [We should use DataCorruptionError for cases where we hit 1Mb limit for op batches](https://dev.azure.com/fluidframework/internal/_workitems/edit/623)

For more information about how to contribute to this repo, visit this [page](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md)

## Description

Today when we reach the 1Mb limit for op batches, we simply trigger a generic error instead of the DataCorruption one we should have. The DataCorruption error will trigger an incident on the FFX and that will be investigated. 

@markfields  - I was talking to Vlad and we need to let the FFX know that we are changing it and the incidents will need to be investigated.  Thoughts ? 

## Steps to Reproduce Bug and Validate Solution

- Submitting a 1Mb op batches 

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [X] All new and existing tests passed.
-   [x] My code follows the code style of this project.
-   [X] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [X] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [X] No

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.

## Testing

> I'll run a load test to make sure if it triggers errors on our end.

### CODE REVIEW

**Review by vladsud — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1747,14 +1748,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this.consecutiveReconnects++;
 
             if (!this.shouldContinueReconnecting()) {
-                this.closeFn(new GenericError(
-                    // pre-0.58 error message: MaxReconnectsWithNoProgress
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1751]: Please leave this comment, it will help someone searching the code for this error message when looking at telemetry from older clients.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -1747,14 +1748,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
+                    DataProcessingError.create(
+                        "Runtime detected too many reconnects with no progress syncing local ops",
+                        "setConnectionState",
+                        undefined,
+                       {
+                        dataLoss: 1,
                         attempts: this.consecutiveReconnects,
                         pendingMessages: this.pendingStateManager.pendingMessagesCount,
-                    }));
+                    }) as DataProcessingError);
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1760]: Why is this needed?

**Review by markfields — APPROVED** [review]

**Comment by markfields** [comment]
#10657 will unblock you in terms of telemetry props being lost.  Ready to merge unless there's any blocking PR feedback.
